ci/jobs/build_clickhouse.py

```diff
@@ -43,9 +52,86 @@ def parse_args():
 -DCMAKE_C_COMPILER={COMPILER} -DCMAKE_CXX_COMPILER={COMPILER_CPP} \
 -DCOMPILER_CACHE={CACHE_TYPE} -DENABLE_BUILD_PROFILING=1 {DIR}"""
 
-# release:          cmake --debug-trycompile -DCMAKE_VERBOSE_MAKEFILE=1 -LA -DCMAKE_BUILD_TYPE=None -DSANITIZE= -DENABLE_CHECK_HEAVY_BUILDS=1 -DENABLE_CLICKHOUSE_SELF_EXTRACTING=1 -DENABLE_TESTS=0 -DENABLE_UTILS=0 -DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON -DCMAKE_INSTALL_PREFIX=/usr -DCMAKE_INSTALL_SYSCONFDIR=/etc -DCMAKE_INSTALL_LOCALSTATEDIR=/var -DCMAKE_SKIP_INSTALL_ALL_DEPENDENCY=ON -DSPLIT_DEBUG_SYMBOLS=ON -DBUILD_STANDALONE_KEEPER=1 -DCMAKE_C_COMPILER=clang-18 -DCMAKE_CXX_COMPILER=clang++-18 -DCOMPILER_CACHE=sccache -DENABLE_BUILD_PROFILING=1 ..
-# binary release:   cmake --debug-trycompile -DCMAKE_VERBOSE_MAKEFILE=1 -LA -DCMAKE_BUILD_TYPE=None -DSANITIZE= -DENABLE_CHECK_HEAVY_BUILDS=1 -DENABLE_CLICKHOUSE_SELF_EXTRACTING=1 -DCMAKE_C_COMPILER=clang-18 -DCMAKE_CXX_COMPILER=clang++-18 -DCOMPILER_CACHE=sccache -DENABLE_BUILD_PROFILING=1 ..
-# release coverage: cmake --debug-trycompile -DCMAKE_VERBOSE_MAKEFILE=1 -LA -DCMAKE_BUILD_TYPE=None -DSANITIZE= -DENABLE_CHECK_HEAVY_BUILDS=1 -DENABLE_CLICKHOUSE_SELF_EXTRACTING=1 -DENABLE_TESTS=0 -DENABLE_UTILS=0 -DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON -DCMAKE_INSTALL_PREFIX=/usr -DCMAKE_INSTALL_SYSCONFDIR=/etc -DCMAKE_INSTALL_LOCALSTATEDIR=/var -DCMAKE_SKIP_INSTALL_ALL_DEPENDENCY=ON -DCMAKE_C_COMPILER=clang-18 -DCMAKE_CXX_COMPILER=clang++-18 -DSANITIZE_COVERAGE=1 -DBUILD_STANDALONE_KEEPER=0 -DCOMPILER_CACHE=sccache -DENABLE_BUILD_PROFILING=1 ..
+CMAKE_BASE = (
+    "cmake --debug-trycompile -DCMAKE_VERBOSE_MAKEFILE=1"
+    " -LA -DENABLE_CHECK_HEAVY_BUILDS=1 -DENABLE_BUILD_PROFILING=1"
+    f" -DCMAKE_C_COMPILER={ToolSet.COMPILER_C} "
+    f"-DCMAKE_CXX_COMPILER={ToolSet.COMPILER_CPP}"
+)
+
+SANITIZERS = {
+    "asan": "address",
+    "msan": "memory",
+    "tsan": "thread",
+    "ubsan": "undefined",
+}
+
+
+def build_args(build_type: List[str]) -> Tuple[str, str]:
+    cmake = CMAKE_BASE
```

Comment:
why not a simple if/elif structure?
this looks error prone when updating, you really have to put in the mental effort to check if you updated everything and fix or add the new build_name and make sure it goes into the right if blocks

Reply:
```
def get_cmake_cmd(build_name):
   if build_name == "amd_release":
       cmake_cmd = "-DCMAKE_BUILD_TYPE=RelWithDebInfo....."
   elif build_name == "amd_asan":
       ........
   else:
      assert False, f"Invalid Build Name {build_name}"
    
   return cmake_cmd
```
      
 it's easy to find how to add new build, change existing one or whatever